rllib/utils/minibatch_utils.py

```diff
@@ -69,13 +70,30 @@ def __iter__(self):
                 n_steps = self._minibatch_size
 
                 samples_to_concat = []
+
                 # cycle through the batch until we have enough samples
-                while n_steps >= len(module_batch) - s:
-                    sample = module_batch[s:]
-                    samples_to_concat.append(sample)
-                    n_steps -= len(sample)
-                    s = 0
-                    self._num_covered_epochs[module_id] += 1
+                if module_batch._slice_seq_lens_in_B:
```

Comment:
Since we prepared the batch to contain seq lens for recurrent RLModuels before, we can now split based upon seq lens.